silq/instrument_interfaces/keysight/E8267D_interface.py

```diff
@@ -188,6 +203,9 @@ def get_additional_pulses(self, connections) -> List[Pulse]:
                                if pulse.frequency_sideband is not None else None
                                for pulse in self.pulse_sequence}
 
+        assert self.FM_mode() != 'ramp' or self.envelope_IQ(), \
+            "Cannot use 'ramp' FM mode while envelope_IQ == False"
+
```

Comment:
what ramp mode has to do with envelope_padding for IQ modulation?

Reply:
added note:
        # Ramp mode does not work with envelope IQ since the output can only be
        # turned off via gating (0V on ramp input corresponds to a pulse at carrier)

Reply:
Is it clear now?

Reply:
yes